clcache.py

```diff
@@ -747,14 +747,32 @@ def _preprocessToStdout(options):
         return ('E' in options or 'EP' in options) and 'P' not in options
 
     @staticmethod
+    def _parsePathArgument(options, argumentName, sourceFile, extension):
+        if argumentName in options:
+            # Handle user input
+            outputFile = options[argumentName][0]
+            outputFile = os.path.normpath(outputFile)
+
+            if os.path.isdir(outputFile):
+                outputFile = os.path.join(outputFile, basenameWithoutExtension(sourceFile) + extension)
+        else:
+            # Generate from .c/.cpp filename
+            outputFile = basenameWithoutExtension(sourceFile) + extension
+        return outputFile
```

Comment:
I think if you rewrite this function slightly such that at the top, you define (and then use)

```python
inferredOutputFile = basenameWithoutExtension(sourceFile) + extension
```

you end up with only this one place where `basenameWithoutExtension` is used in this file, in which case it may be interesting to just inline the function here, i.e. define

```python
inferredOutputFile = os.path.splitext(os.path.basename(sourceFile))[0] + extension
```

...and then move the `basenameWithoutExtension` function out into `unittests.py`.